var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,341 @@
 
+    # ==========================================================================
+    # Variants
+    # ==========================================================================
+
+    # SUNDIALS solvers
+    sun_solvers = ['CVODE', 'CVODES', 'ARKODE', 'IDA', 'IDAS', 'KINSOL']
+
+    for pkg in sun_solvers:
+        variant(pkg, default=True,
+                description='Enable %s solver' % pkg)
+
+    # Real type
+    variant(
+        'precision',
+        default='double',
+        description='''real type precision''',
+        values=('single', 'double', 'extended'),
+        multi=False
+    )
+
+    # Index type
+    variant(
+        'indextype',
+        default='int64_t',
+        description='''index integer type''',
+        values=('int32_t', 'int64_t'),
+        multi=False
+    )
+
+    # Parallelism
     variant('mpi',     default=True,
-            description='Enable MPI parallelism')
-    variant('lapack',  default=True,
-            description='Use external BLAS/LAPACK libraries')
-    variant('klu',     default=False,
+            description='Enable MPI parallel vector')
+    variant('openmp',  default=False,
+            description='Enable OpenMP parallel vector')
+    variant('pthread', default=False,
+            description='Enable Pthreads parallel vector')
+    variant('cuda',    default=False,
+            description='Enable CUDA parallel vector')
+    variant('raja',    default=False,
+            description='Enable RAJA parallel vector')
+
+    # External libraries
+    variant('lapack',     default=False,
+            description='Enable LAPACK direct solvers')
+    variant('klu',        default=False,
             description='Enable KLU sparse, direct solver')
-    variant('superlu', default=False,
+    variant('superlu-mt', default=False,
             description='Enable SuperLU_MT sparse, direct solver')
-    variant('openmp',  default=False,
-            description='Enable OpenMP parallelism')
-    variant('pthread', default=True,
-            description='Enable Pthreads parallelism')
-    variant('hypre', default=False,
-            description='Enable Hypre parallel vector for MPI parallelism')
-
-    depends_on('cmake',              type='build')
-    depends_on('mpi',                when='+mpi')
-    depends_on('mpi',                when='@2.7:+hypre')
-    depends_on('hypre',              when='@2.7:+hypre')
-    depends_on('blas',               when='+lapack')
-    depends_on('lapack',             when='+lapack')
-    depends_on('suite-sparse',       when='+klu')
-    depends_on('superlu-mt+openmp',  when='+superlu+openmp')
-    depends_on('superlu-mt+pthread', when='+superlu+pthread')
-
-    def install(self, spec, prefix):
+    variant('hypre',      default=False,
+            description='Enable Hypre MPI parallel vector')
+    variant('petsc',      default=False,
+            description='Enable PETSc MPI parallel vector')
+
+    # Library type
+    variant('shared', default=True,
```

Comment:
FYI, there is currently a discussion in Spack on how to unify static/shared https://github.com/LLNL/spack/issues/5269 If you have any preferences where that go, please feel free to join the discussion